### PR TITLE
Remove FromGAC & OnTPAList

### PR DESCRIPTION
**Pull request by sdmaclea**

### CODE REVIEW

File: src/vm/assemblynative.cpp [hunk 1]
```diff
@@ -1296,40 +1296,37 @@ INT_PTR QCALLTYPE AssemblyNative::GetLoadContextForAssembly(QCall::AssemblyHandl
    
     // Platform assemblies are semantically bound against the "Default" binder which could be the TPA Binder or
     // the overridden binder. In either case, the reference to the same will be returned when this QCall returns.
-    if (!pPEAssembly->IsProfileAssembly())
```
Comment (sdmaclea) [inline src/vm/assemblynative.cpp]: If IsGAC wasn't lying, this should have been equivalent to `if (!false)`

If IsGAC was lying this seems like it still should be OK as `if (!AreSameBinderInstance(pTPABinder, pOpaqueBinder))` below should handle this being incorrect.

File: src/vm/coreassemblyspec.cpp [hunk 2]
```diff
@@ -181,38 +181,15 @@ VOID  AssemblySpec::Bind(AppDomain      *pAppDomain,
...
-    pResult->SetHRBindResult(hr);
-    if (SUCCEEDED(hr))
-    {
-        BOOL fIsInGAC = FALSE;
-        BOOL fIsOnTpaList = FALSE;
-
-        // Only initialize TPA/GAC status if we bound using DefaultContext
-        if (fBoundUsingTPABinder == true)
-        {
-            fIsInGAC = pAppDomain->IsImageFromTrustedPath(result->GetNativeOrILPEImage());
```
Comment (sdmaclea) [inline src/vm/coreassemblyspec.cpp]: Looks like `IsImageFromTrustedPath` can also be removed.

File: src/vm/coreassemblyspec.cpp [hunk 3]
```diff
@@ -181,38 +181,15 @@ VOID  AssemblySpec::Bind(AppDomain      *pAppDomain,
...
-            const SString &sImagePath = result->GetNativeOrILPEImage()->GetPath();
-            if (pTPABinder->IsInTpaList(sImagePath))
```
Comment (sdmaclea) [inline src/vm/coreassemblyspec.cpp]: Also `IsInTpaList` can be removed

File: src/vm/assemblynative.cpp [hunk 4]
```diff
@@ -1296,40 +1296,37 @@ INT_PTR QCALLTYPE AssemblyNative::GetLoadContextForAssembly(QCall::AssemblyHandl
    
     // Platform assemblies are semantically bound against the "Default" binder which could be the TPA Binder or
     // the overridden binder. In either case, the reference to the same will be returned when this QCall returns.
```
Comment (jkotas) [inline src/vm/assemblynative.cpp]: You can delete "or the overriden binder" part. It does not exist anymore (you have deleted last bit of it in the last PR).

**Comment by jkotas** [comment]
Looks like there is also a build break to fix.

**Pull request merged**